src/components/PersonComponent/PersonComponent.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FC } from 'react';
+import { PersonLink } from '../PersonLink';
+import classNames from 'classnames';
+import { Person } from '../../types';
+import { useParams } from 'react-router-dom';
+
+interface Props {
+  people: Person[];
+}
+
+export const PersonComponent: FC<Props> = ({ people }) => {
+  const findPersonByName = (name: string | undefined) => {
+    return people.find(person => person.name === name) || null;
+  };
+
+  const actualSlug = useParams().slug;
+
+  const getPersonClassNames = (slug: string) =>
+    classNames({
+      'has-background-warning': slug === actualSlug,
+    });
+
+  return (
+    <>
+      {people.map(person => {
+        const { sex, born, died, motherName, fatherName, slug } = person;
+
+        const mother = findPersonByName(motherName || undefined);
+        const father = findPersonByName(fatherName || undefined);
```

Comment:
- You can prepare people when you just load them from the server
- And it doesn't make sense to send undefined if it's already undefined. `|| undefined` - is redundant